lib/gc.cpp

```diff
@@ -56,10 +56,17 @@ extern "C" size_t GC_get_heap_size(void);
 extern "C" int GC_print_stats;
 
 static void gc_callback() {
-    size_t count;
-    LOG1("****** GC called ****** (heap size " << GC_get_heap_size() << ")");
-    LOG2("cstring cache size " << cstring::cache_size(count) << " (count=" << count << ")");
-    GC_print_stats = LOGGING(2) ? 1 : 0;  // unfortunately goes directly to stderr!
+    if (Log::verbose())
```

Comment:
This subtljy changes the logging behavior, basing it off of the global -v level rather than the -Tgc option, which defaults to one less than the -v level.  So you've changed these to require one less -v option to be triggered.  Will using the `LOGGING` macro run into the same memory allocation issues?